src/pages/ReimbursementAccount/BankAccountPlaidStep.js

```diff
@@ -37,27 +38,28 @@ class BankAccountPlaidStep extends React.Component {
     }
 
     submit() {
-        const selectedPlaidBankAccount = this.props.plaidData.selectedPlaidBankAccount;
-        if (!selectedPlaidBankAccount) {
-            return;
-        }
+        const selectedPlaidBankAccount = _.findWhere(lodashGet(this.props.plaidData, 'bankAccounts', []), {
+            plaidAccountID: ReimbursementAccountUtils.getDefaultStateForField(this.props, 'plaidAccountID'),
+        });
 
-        ReimbursementAccount.updateReimbursementAccountDraft({
+        const bankAccountData = {
             routingNumber: selectedPlaidBankAccount.routingNumber,
             accountNumber: selectedPlaidBankAccount.accountNumber,
             plaidMask: selectedPlaidBankAccount.mask,
             isSavings: selectedPlaidBankAccount.isSavings,
-            bankName: selectedPlaidBankAccount.bankName,
+            bankName: this.props.plaidData.bankName,
             plaidAccountID: selectedPlaidBankAccount.plaidAccountID,
-            plaidAccessToken: selectedPlaidBankAccount.plaidAccessToken,
-        });
+            plaidAccessToken: this.props.plaidData.plaidAccessToken,
+        };
+        ReimbursementAccount.updateReimbursementAccountDraft(bankAccountData);
```

Comment:
NAB we can clean this up later, but I don't think that we need this since Form automatically updates the draft data if we pass `shouldSaveDraft` to each input

```suggestion
        ReimbursementAccount.updateReimbursementAccountDraft(bankAccountData);
```

Reply:
Okay, cc @nkuoch and @MariaHCD so you're aware. You're saying that we could just remove this as long as we pass `shouldSaveDraft` to each form component below, yeah?